### PR TITLE
NAS-132125 / 25.04 / Replace iscsi.discoveryauth.* api

### PR DESCRIPTION
**Pull request by bmeagherix**
Earlier in FT there was a PR #14212 that introduced a **new** `iscsi.discoveryauth.*` CRUD API - to reflect the reality of SCALE's iSCSI discovery auth (vs CORE).

After discussing further with the UI team, it was decided to **simplify** the (collective) API, by instead folding this functionality into the regular `iscsi.auth.*` CRUD API as an additional attribute `discovery_auth`.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/iscsi_/portal.py [hunk 1]
```diff
@@ -69,17 +65,8 @@ async def config_extend(self, data, context):
...
-        # )
-        # data['discovery_authgroup'] = data.pop('discoveryauthgroup')
-        auths = await self.middleware.call('iscsi.discoveryauth.query')
-        if auths:
-            data['discovery_authmethod'] = auths[0]['authmethod']
-            data['discovery_authgroup'] = auths[0]['authgroup']
-        else:
-            data['discovery_authmethod'] = "NONE"
-            data['discovery_authgroup'] = None
+        data['discovery_authmethod'] = "NONE"
```
Comment (anodos325) [inline src/middlewared/middlewared/plugins/iscsi_/portal.py:68]: `IscsiAuthType.NONE` ?

File: src/middlewared/middlewared/plugins/iscsi_/utils.py [hunk 2]
```diff
@@ -1,10 +1,18 @@
+from enum import StrEnum
+
 import bidict
 
 
+class IscsiAuthType(StrEnum):
```
Comment (anodos325) [inline src/middlewared/middlewared/plugins/iscsi_/utils.py:6]: Is it a problem that this class name collides with IscsiAuthType in `api/base/types/iscsi.py`?

Comment (bmeagherix) [reply src/middlewared/middlewared/plugins/iscsi_/utils.py:6]: I don't believe this is a problem.  The one in `api/base/types/iscsi.py` is only used for API definitions.

Comment (anodos325) [reply src/middlewared/middlewared/plugins/iscsi_/utils.py:6]: Maybe `IscsiAuthTypeEnum`?

Comment (anodos325) [reply src/middlewared/middlewared/plugins/iscsi_/utils.py:6]: Okay.

**Review by anodos325 — APPROVED** [review]

**Pull request merged**